File: app/src/main/java/com/mahesh/mvpsample/main/MainContract.java
package com.mahesh.mvpsample.main;

import com.mahesh.mvpsample.BasePresenter;

public class MainContract {

    interface View {

        void openDetailScreen();
    }

    interface Presenter extends BasePresenter {
        void resume(View view);

        void pause();

        void clickMeButtonClicked();
    }

    interface ModelInteractor {

    }
}

File: app/src/main/java/com/mahesh/mvpsample/main/MainContract.kt
package com.mahesh.mvpsample.main

import com.mahesh.mvpsample.BasePresenter

class MainContract {
    interface View {
        fun openDetailScreen()
    }

    internal interface Presenter : BasePresenter {
        fun resume(view: View?)
        fun pause()
        fun clickMeButtonClicked()
    }

    internal interface ModelInteractor
}
